src/ai_hawk/linkedIn_easy_applier.py

```diff
@@ -157,7 +158,7 @@ def job_apply(self, job: Job):
             
             # Todo: add this job to skip list with it's reason
             if not self.gpt_answerer.is_job_suitable():
-                return
+                raise JobNotSuitableException("LLM score determined this job is not suitable for you")
```

Comment:
tranferring job not suitable in Exception message is better approach, otherwise we end up in having too many exception classes.

we need separte execption classes when flow is differnt

Reply:
Can you please write you comment in clear, concise message? I don't really understand what you're trying to say. 

Are you suggesting I raise a generic exception `raise Exception("Job Not Suitable)`? This way I dont need to create a custom exception at all? If that's the case, when catching this exception in [`job_manager.py`](https://github.com/49Simon/Auto_Jobs_Applier_AIHawk/blob/6ad0119b3ab677fe62f3a143198e1aa4088b6f2a/src/ai_hawk/job_manager.py#L419) it will be caught by the generic `Exception` and save it to `failed.json`.

If we have a custom exception, that inherits from JobSkipException (your above comment) then we can catch the exception and gracefully handle it by saving to `skipped.json` with its reason. So what exactly are you suggesting I do?

Reply:
sorry my bad. not conveying it properly. updated the message